File: src/AC/066.cpp
// LeetCode: https://leetcode.com/problems/plus-one/

#include <vector>

class Solution {
public:
    std::vector<int> plusOne(std::vector<int>& digits) {
        for (int i=digits.size()-1; i>=0; i--) {
            if (digits[i] == 9) {
                if (i == 0) {
                    digits.insert(digits.begin(), 1);
                    digits[i+1] = 0;
                    digits[i] = 1;
                    break;
                }

                digits[i] = 0;
                continue;
            }

            digits[i]++;
            break;
        }

        return digits;
    }
};

File: src/AC/066/066.rs
impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        if digits.len() == 0 {
            return vec![1];
        }

        let mut digits_copy = digits.clone();
        for i in (0..digits_copy.len()).rev() {
            if digits_copy[i] == 9 {
                digits_copy[i] = 0;
            } else {
                digits_copy[i] += 1;
                return digits_copy;
            }
        }

        digits_copy.insert(0, 1);
        digits_copy
    }
}
